integrations/ad_reader.py

```diff
@@ -0,0 +1,180 @@
+import os
+import json
+from winrm import Session
+
+WINRM_HOST = os.environ.get('WINRM_HOST', None)
+AD_SYSTEM_USER = os.environ.get('AD_SYSTEM_USER', None)
+AD_PASSWORD = os.environ.get('AD_PASSWORD', None)
+SCHOOL_AD_SYSTEM_USER = os.environ.get('SCHOOL_AD_SYSTEM_USER', None)
+SCHOOL_AD_PASSWORD = os.environ.get('SCHOOL_AD_PASSWORD', None)
+# SEARCH_BASE
+# PROPERTIES
+# SKIP_BRUGERTYPE
+
+
+class ADParameterReader(object):
+
+    def __init__(self):
+        self.session = Session(
+            'http://' + WINRM_HOST + ':5985/wsman',
+            transport='kerberos',
+            auth=(None, None)
+        )
+        self.results = {}
+
+    def _run_ps_script(self, ps_script):
+        """
+        Run a power shell script and return the result. If it fails, the
+        error is returned in its raw form.
+        :param ps_script: The power shell script to run.
+        :return: A dictionary with the returned parameters.
+        """
+        r = self.session.run_ps(ps_script)
+        if r.status_code == 0:
+            if r.std_out:
+                response = json.loads(r.std_out.decode('Latin-1'))
+            else:
+                response = {}
+        else:
+            response = r.std_err
+        return response
+
```

Comment:
Jeg ville foretrække at få password og brugernavn sendt ind i metoden i stedet for at afgøre derinde, hvad man skal bruge, eller måske som attributter på self.